deploy/kubernetes/elastic-agent-standalone/elastic-agent-standalone-daemonset.yaml

```diff
@@ -141,3 +143,7 @@ spec:
         - name: sys-kernel-debug
           hostPath:
             path: /sys/kernel/debug
+        - name: elastic-agent-state
+          hostPath:
+            path: /var/lib/elastic-agent-standalone/kube-system/state
```

Comment:
Is /kube-system folder access related to the fact that we run daemonset as root? (https://github.com/elastic/elastic-agent/blob/efd3fa4e16fb932669f035d61a0f14db05ecc3ac/deploy/kubernetes/elastic-agent-managed-kubernetes.yaml#L66) 

Dont know what happens if we are not root users and we need access to this folder.

Reply:
Another question is that default state path in agent is : `"/usr/share/elastic-agent/state`. ([link](https://github.com/elastic/beats/pull/24817/files#diff-1f8e1e2079524931fa8f3cf121667a25baa2b998557136500d5e7596b46e9187R44))

Do you think it makes sense to allign them?

Reply:
> Do you think it makes sense to allign them?

`/var/lib/elastic-agent-standalone/kube-system/state` is the path on the kubernetes node, this will be mounted to the `"/usr/share/elastic-agent/state"` inside the elastic-agent Pod - https://github.com/elastic/elastic-agent/pull/2550/files#diff-7ea6944c2361d6182f8906930f7779f59c5bae5400cdafbb71a7f1c12e8d7e87R110

Reply:
> Is /kube-system folder access related to the fact that we run daemonset as root?

no, it is just to keep namespace name in the folder path
> Dont know what happens if we are not root users and we need access to this folder.

at least for `hostPath` I believe it will not work - volume will be mounted with the root as an owner, and not root elastic-agent process will not be able to write to this folder in this case
As an option we can consider using the local volume - https://kubernetes.io/blog/2019/04/04/kubernetes-1.14-local-persistent-volumes-ga/#how-is-it-different-from-a-hostpath-volume:
```
Additional benefits include support for formatting of block devices during mount, and volume ownership using fsGroup.
```
I didn't test it, not sure how/if DaemonSet in combination with this statement would work togetger: `Kubernetes scheduler ensures that a pod using a Local Persistent Volume is always scheduled to the same node`